packages/nimble-components/src/wafer-map/modules/rendering.ts

```diff
@@ -76,18 +61,22 @@ export class RenderingModule {
         transformedCanvasMinPoint[0] -= dieWidth;
         transformedCanvasMinPoint[1] -= dieHeight;
 
-        for (const die of this.dies) {
-            if (
-                this.isDieVisible(
-                    die,
-                    transformedCanvasMinPoint,
-                    transformedCanvasMaxPoint
-                )
-            ) {
-                context.fillStyle = die.fillStyle;
-                context.fillRect(die.x, die.y, dieWidth, dieHeight);
+        Object.entries(this.wafermap.dataManager.renderInfo).forEach(
+            ([fillStyle, dies]) => {
+                context.fillStyle = fillStyle;

```

Comment:
Just so I have a clear understanding of the root cause of the issue, the `context.fillStyle` set is what I presume is the bottleneck?

Reply:
After the tests I did we will postpone this improvement and discuss with my team how to move forward.

I think we will revisit in a week or two